### ISSUE TITLE
Login Failure

### ISSUE THREAD

**Issue by alastaid** [blakadder/tuya-uncover#4]
This could be my misunderstanding (so apologies in advance), using the script thus:

python3 uncover.py -v smartlife xx@xx.com xxx

I get "Incorrect Account ID or Password", if I use:

python3 uncover.py -v tuya xx@xx.com xxx

I get "SING_VALIDATE_FAILED_4", I am using the Tuya Smart app on android, and inputting the email and password as I use in the app for credentials in the Python script. Am I using this correctly?  Thanks

**Comment by Bhoft** [comment]
do you have any special characters in your password?
if yes see my remark in this issue https://github.com/blakadder/tuya-uncover/issues/5

**Comment by latetedemelon** [comment]
Unfortunately I have to report the same issue.  I have special characters (?!) in my password, and while quotes or hardcoding the password allows the program to run (@Bhoft - #5) I still get the same result as @alastaid.

I've got generic devices and setting the vendor to smartlife gives me "Invalid username or password" and setting it to tuya gives me  "SING_VALIDATE_FAILED_4".

I've got the cloud set up and working and can see my devices in both my app and in my Home Assistance Tuya CLOUD integration.  

@blakadder - Any hints please?

**Comment by blakadder** [comment]
Probably changes with their login system. There should be other options to grab the local keys for those provides

**Comment by szupi-ipuzs** [comment]
I can see that the keys for birdlover app that used to work for me are now being rejected with the same error (SING_VALIDATE_FAILED_4). It probably means they deprecated the old keys or completely reworked the way some OEM apps access the cloud. I've seen the new apps are using some framework called "ThingClips" - it's probably related.

**Comment by philhar88** [comment]
Is there a workaround for this? new client_id and secret values perhaps? I've tried using the tuya vendor and have the same SING_VALIDATE_FAILED_4 error

**Comment by phpony** [comment]
Yep, still doesn't work with Tuya Home accounts. SING_VALIDATE_FAILED_4 error.
